lib/spack/spack/package.py

```diff
@@ -380,6 +380,22 @@ def __init__(self, spec):
         if not hasattr(self, 'list_depth'):
             self.list_depth = 1
 
+        # Set default licensing information
+        if not hasattr(self, 'license_required'):
+            self.license_required = False
+
+        if not hasattr(self, 'license_comment'):
+            self.license_comment = '#'
+
+        if not hasattr(self, 'license_files'):
+            self.license_files = []
+
+        if not hasattr(self, 'license_vars'):
+            self.license_vars = []
+
+        if not hasattr(self, 'license_url'):
+            self.license_url = None
```

Comment:
Hi, just a style question : why are you **conditionally** setting attributes on `Package`? Wouldn't just setting them work?